libraries/chain/database.cpp

```diff
@@ -1244,20 +1244,26 @@ asset create_vesting2( database& db, const account_object& to_account, asset liq
       if( liquid.symbol.space() == asset_symbol_type::smt_nai_space )
       {
          FC_ASSERT( liquid.symbol.is_vesting() == false );
-         // Get share price.
-         const auto& smt = db.get< smt_token_object, by_symbol >( liquid.symbol );
-         FC_ASSERT( smt.allow_voting == to_reward_balance, "No voting - no rewards" );
-         price vesting_share_price = to_reward_balance ? smt.get_reward_vesting_share_price() : smt.get_vesting_share_price();
+
+         const auto& token = db.get< smt_token_object, by_symbol >( liquid.symbol );
+
+         if ( to_reward_balance )
+            FC_ASSERT( token.allow_voting == true, "Cannot create vests for the reward balance when voting is disabled." );
```

Comment:
This assertion should never fail as voting with a token that doesn't allow voting should not be allowed further up in the chain. As it turns out, it seems I forgot that assertion in my work on #2747

Can you add to the message that this assertion should not be triggered, so that if it is, it raises an alarm much faster?